server/utils/limited_readcloser.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package utils
+
+import (
+	"io"
+)
+
+type LimitReadCloser struct {
```

Comment:
Is there some reason for not using `io.LimitReader`? Or at least incorporating it into the implementation here?

Reply:
It seemed simpler to cut-paste from it than to wrap around, but I'll give it a try to see if it simplifies the code. I agree that the "Limit" functionality does not belong in the `expvar` package, it should know nothing about it.

Reply:
done